doc/ale-supported-languages-and-tools.txt

```diff
@@ -338,6 +338,8 @@ Notes:
   * `languageserver`
   * `puppet`
   * `puppet-lint`
+* Purescript
+  * `purels`
```

Comment:
Name this and the heading `purescript-language-server`. Update the table of contents in `ale.txt` for the options for linters. Update the Markdown list of supported tools.